packages/flutter/lib/src/widgets/gesture_detector.dart

```diff
@@ -924,90 +932,37 @@ class RawGestureDetectorState extends State<RawGestureDetector> {
     return widget.child == null ? HitTestBehavior.translucent : HitTestBehavior.deferToChild;
   }
 
-  void _handleSemanticsTap() {
-    final TapGestureRecognizer recognizer = _recognizers[TapGestureRecognizer];
-    assert(recognizer != null);
-    if (recognizer.onTapDown != null)
-      recognizer.onTapDown(TapDownDetails());
-    if (recognizer.onTapUp != null)
-      recognizer.onTapUp(TapUpDetails());
-    if (recognizer.onTap != null)
-      recognizer.onTap();
+  void _assignSemantics(RenderSemanticsGestureHandler renderObject) {
+    if (_semanticsMapping == null)
+      return;
+    renderObject
```

Comment:
I think we should factor out the whole of _assignSemantics into the delegate, rather than each individual part. There's various advantages of doing that; one of them, in particular, is that we can add new handler types without breaking all the existing delegates.

Reply:
If I understand correctly you want to break up `_assignSemantics` into individual kinds. I think it's unnecessary because both `_assignSemantics` and `_GestureSemantics` are private.

Reply:
I mean that this would be the delegate's API:

```dart
abstract class GestureSemanticsMapping {
  const GestureSemanticsMapping();
  void updateSemantics(RenderSemanticsGestureHandler renderObject);
}
```

Reply:
...and the default would be:

class _DefaultGestureSemanticsMapping extends GestureSemanticsMapping {
  _DefaultGestureSemanticsMapping(this.owner);
  final RawGestureDetectorState owner;

  @override
  void updateSemantics(RenderSemanticsGestureHandler renderObject) {
    // do things with owner._recognizers to fill in renderObject...
  }
}

Reply:
(the actual default for the widget would be null, and the `_DefaultGestureSemanticsMapping` would get created during `initState` / `didChangeWidget`. Or else we could just forgo the default and hard-code the default behaviour without using the delegate, though that might make the code a bit uglier.)